rust/operator-binary/src/airflow_controller.rs

```diff
@@ -847,6 +827,51 @@ fn build_rolegroup_metadata(
     Ok(metadata)
 }
 
+pub fn build_group_listener(
+    airflow: &v1alpha1::AirflowCluster,
+    resolved_product_image: &ResolvedProductImage,
+    rolegroup: &RoleGroupRef<v1alpha1::AirflowCluster>,
+    listener_class: String,
+) -> Result<Listener> {
+    Ok(Listener {
+        metadata: ObjectMetaBuilder::new()
+            .name_and_namespace(airflow)
+            .name(airflow.group_listener_name(rolegroup))
+            .ownerreference_from_resource(airflow, None, Some(true))
+            .context(ObjectMissingMetadataForOwnerRefSnafu)?
+            .with_recommended_labels(build_recommended_labels(
+                airflow,
+                AIRFLOW_CONTROLLER_NAME,
+                &resolved_product_image.app_version_label,
+                &rolegroup.role,
+                &rolegroup.role_group,
+            ))
+            .context(ObjectMetaSnafu)?
+            .build(),
+        spec: ListenerSpec {
+            class_name: Some(listener_class),
+            ports: Some(listener_ports()),
+            ..ListenerSpec::default()
+        },
+        status: None,
+    })
+}
+
+fn listener_ports() -> Vec<ListenerPort> {
+    vec![
+        ListenerPort {
+            name: METRICS_PORT_NAME.to_string(),
+            port: METRICS_PORT.into(),
+            protocol: Some("TCP".to_string()),
+        },
```

Comment:
Don't think we need metrics here?

Reply:
Removed from listener ports: cb85a6a

Reply:
I would prefer to keep them. Customers aren't necessarily using Prometheus within k8s, although that's the probably best solution. *If* we remove them this should be marked as breaking change in the changelog IMHO.
Maybe worth asking customers if anyone relies on this functionality before removing it.

Reply:
If we support metrics-via-listener *now*, when introducing it, then it will be more difficult to drop support for it later on. Presumably users who are currently exposing metrics externally can continue to do so (using port-forwards, manually resolved node-IPs etc.)

Reply:
I guess this smells like a decision, if that is ok for you?

Reply:
I thought we sort of already decided this last Thursday (with 5-6 of us) when we mentioned removing the metrics ports/addresses from the listener-op.

Reply:
Hmm yeah, I can see your point.
I can't really recall it 100%, so I'm not sure, but I *thought* that at least I was not happy with it.
But I'm happy to create a decision and not put this on you. Gathering data about customer usage would be part of it I guess.

Reply:
That's fine by me. Maybe after a daily so we don't have to wait too long.

Reply:
See https://github.com/stackabletech/decisions/issues/51